AirSkyBoat/AirSkyBoat
# [core] Implement Pre-Neo-Dynamis Zones, Functions, Mobs

**_I affirm:_**

## What does this pull request do?
+ Added spawning tables to each Dynamis zone to dictate which mobs to spawn.
+ Removed all references to IDs within Dynamis globals to allow for the use of dynamic entities.
+ Created dynamic entity spawning scripts for NMs, Nightmare Mobs, Statues, Normal Mobs, and Pets.
+ Rebalanced the stat lines for all of the above mobs by removing any non-sourced reductions in stat lines or additional mods/effects.
+ Implemented dynamic methods of modifying all Trail Markings and Herioglyphics
+ Modulized all mob functions, skill lists, spell lists, and drop tables.
+ Dynamically allocates appropriate mob functions, skills, spells, and drops during the instantiation of dynamic entities.
+ Implemented core dynamis functions for hourglass creation, management, duplication, and removal.
+ Implemented core functions for managing the zone using onZoneTick().
+ Implemented core functionality to NPCs for hourglass handling, starting the dynamis zone, providing the initial hourglass, accepting an hourglass, and transporting the player.
+ Implemented updated mechanics to allow for QMs in dynamis zones to function properly and function in a more reasonable manner related to server performance.
+ Implemented SJ restriction for dreamlands zones which accurately removes buffs upon entry compared to previous versions.
+ Implemented all missing mobskills.
+ Created seperate function luas for each zone which house all dynamic functions for a mob.
+ Added special spawn animation to every mob that requires it. Does not apply to mobs without a parent.
+ Setup renamer lists for Zach's addon.

## Steps to test these changes
+ Tested each zone rigorously for mechanic usage by NMs and Bosses.
+ Tested the spawning mechanism for normal mobs, NMs, statues, 
+ Tested all NMs and Bosses in each zone for functionality and accuracy.
+ Tested spawning systems with potential nullptrs and nil values.
+ Tested mob luas to ensure accuracy.
+ Tested all waves in each zone to ensure spawning.
+ Tested all defeat requirements.
+ Tested all variable setting within each zone.
+ Tested mob stat distribution and confirmed it is working as intended.
